### PR TITLE
test: data source resources integration tests

### PR DESCRIPTION
**Pull request by hatimkhan**
Issue #: GALI-736

This PR includes the tests for the data source resources that are introduced as part of BYOB.

Checklist:

- [X] Have you successfully tested with your changes locally?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: main/integration-tests/__test__/api-tests/data-sources/register-ds-bucket.test.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+  beforeAll(async () => {
+    setup = await runSetup();
+    adminSession = await setup.defaultAdminSession();
+    accountId = setup.gen.accountId();
+    await adminSession.resources.dataSources.accounts.create({ id: accountId });
+  });
+
+  afterAll(async () => {
+    await setup.cleanup();
```
Comment (SanketD92) [inline main/integration-tests/__test__/api-tests/data-sources/register-ds-bucket.test.js:32]: Could you confirm cleanup is working as expected? (I was seeing some leftover resources on the DDB post-test last time I ran these tests)

File: main/integration-tests/__test__/api-tests/data-sources/register-ds-bucket.test.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    it('should return bucket registration information if admin', async () => {
+      const admin2Session = await setup.createAdminSession();
+      const name = setup.gen.string({ prefix: 'ds-bucket-test' });
+
+      await expect(
+        admin2Session.resources.dataSources.accounts
+          .account(accountId)
+          .buckets()
+          .create({ name }),
+      ).resolves.toHaveProperty('name', name);
```
Comment (SanketD92) [inline main/integration-tests/__test__/api-tests/data-sources/register-ds-bucket.test.js]: Could you also add the accountId check here?
```
.resolves.toMatchObject({name, accountId});
```

File: main/integration-tests/__test__/api-tests/data-sources/register-ds-study.test.js [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+      };
+
+      await expect(
+        admin2Session.resources.dataSources.accounts
+          .account(accountId)
+          .buckets()
+          .bucket(bucketName)
+          .studies()
+          .create(study),
+      ).resolves.toHaveProperty('id', id);
```
Comment (SanketD92) [inline main/integration-tests/__test__/api-tests/data-sources/register-ds-study.test.js]: Could you also add the check for bucket and accountId?
```
.resolves.toMatchObject({id, accountId, bucket: bucketName});
```

File: main/integration-tests/support/resources/data-sources/buckets.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    const gen = this.setup.gen;
+    const accountId = this.accountId;
+    const region = 'us-east-1' || bucket.region;
+
+    return {
+      accountId,
+      name: gen.string({ prefix: 'ds-bucket-test' }),
+      region,
+      awsPartition: 'aws',
+      kmsArn: `arn:aws:kms:${region}:${accountId}:key/2e3c97b6-8bb3-4cf8-bc77-d56ebf847d23`,
```
Comment (SanketD92) [inline main/integration-tests/support/resources/data-sources/buckets.js]: optional: Could you update this string to contain the runId instead?
```
kmsArn: `arn:aws:kms:${region}:${accountId}:key/${gen.string()}-key`,
```
(I can do this in my next PR as well)

File: main/integration-tests/__test__/api-tests/data-sources/register-ds-bucket.test.js (hunk 1 above)
Comment (hatimkhan) [reply main/integration-tests/__test__/api-tests/data-sources/register-ds-bucket.test.js:32]: Yes, needed to do that as part of writing the cleanup logic for the data source. I see all the necessary entries in the dsAccounts and RoleAllocations tables being correctly cleaned up.

File: main/integration-tests/support/resources/data-sources/buckets.js (hunk 4 above)
Comment (hatimkhan) [reply main/integration-tests/support/resources/data-sources/buckets.js]: Done

File: main/integration-tests/__test__/api-tests/data-sources/register-ds-bucket.test.js (hunk 2 above)
Comment (hatimkhan) [reply main/integration-tests/__test__/api-tests/data-sources/register-ds-bucket.test.js]: Done

**Review by SanketD92 — APPROVED** [review]
LGTM

**Pull request merged**